src/test/java/main/HttpMethodsTest.java

```diff
@@ -0,0 +1,151 @@
+package main;
+
+import io.restassured.RestAssured;
+import io.restassured.http.Method;
+import io.restassured.response.Response;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashMap;
+
+import static io.restassured.RestAssured.given;
+import static org.testng.Assert.assertEquals;
+
+public class HttpMethodsTest {
+  @BeforeClass
+  public final void setBaseUriPath() {
+    RestAssured.baseURI = "https://weather.api.here.com";
+    RestAssured.basePath = "/weather/1.0/report.json";
+  }
+
+  @DataProvider(name = "dataSet")
+  public static Object[][] setApiInfo() {
+    HashMap<String, String> basicData = new HashMap<String, String>();
```

Comment:
since basic data is the same for all the tests you can pull it up as class variable (inititilize in beforeClass method) or even constant. It doesn't have to be part of dataProvider since it's always the same